apps/heft/src/plugins/MissingPluginWarningPlugin/JestWarningPlugin.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+
+import { HeftConfiguration } from '../../configuration/HeftConfiguration';
+import { HeftSession } from '../../pluginFramework/HeftSession';
+import { ITestStageContext } from '../../stages/TestStage';
+import { MissingPluginWarningPluginBase } from './MissingPluginWarningPluginBase';
+
+const PLUGIN_NAME: string = 'jest-warning-plugin';
```

Comment:
`[jest-warning-plugin]` seems a bit silly.  Not every JavaScript function needs to be its own Heft plugin.

Why not simply have one `ProjectValidationPlugin` and then the warnings can be like:
```
[project-validation] You have a jest.config.json file, but you're missing the Jest plugin.
[project-validation] You have a webpack.config.js file, but you're missing the Webpack plugin.
[project-validation] I noticed that your heft.json is using some obsolete settings. Here's how to fix that.
```

The main reason we should be making new plugins is to solve problems where a bunch of NPM dependencies and code should not get loaded unless a feature is actually needed.

Plugins have a price:
- We have to write website docs explaining each plugin
- Users have to keep the plugins straight when troubleshooting problems
- Plugins make debugging harder; control flow is nonlinear and it's hard to determine when the taps will fire
- It is difficult for plugins to talk to each other, so they discourage intelligent interactions between components

We should pay this price when there is a clear benefit.